### PR TITLE
Azure: migrate windows job config to pod utilities

### LINKED ISSUE #16132: Azure: upgrade job configurations to PodUtils
**What should be cleaned up or changed**:

- As bootstrap.py is depracted, we should switch all Azure-related job configs to [PodUtils](https://github.com/kubernetes/test-infra/blob/master/prow/pod-utilities.md) ([Motivation of switching](https://github.com/kubernetes/test-infra/blob/master/prow/pod-utilities.md#why-use-pod-utilities)):
  - [x] [cloud-provider-azure-config.yaml](https://github.com/kubernetes/test-infra/blob/master/config/jobs/kubernetes-sigs/cloud-provider-azure/cloud-provider-azure-config.yaml)
  - [x] [sig-azure-config.yaml](https://github.com/kubernetes/test-infra/blob/master/config/jobs/kubernetes/sig-cloud-provider/azure/sig-azure-config.yaml)
  - [x] [release-1.15.yaml](https://github.com/kubernetes/test-infra/blob/master/config/jobs/kubernetes/sig-cloud-provider/azure/release-1.15.yaml)
  - [x] [release-1.16.yaml](https://github.com/kubernetes/test-infra/blob/master/config/jobs/kubernetes/sig-cloud-provider/azure/release-1.16.yaml)
  - [x] [release-1.17.yaml](https://github.com/kubernetes/test-infra/blob/master/config/jobs/kubernetes/sig-cloud-provider/azure/release-1.17.yaml)
  - [x] [azuredisk-csi-driver-config.yaml](https://github.com/kubernetes/test-infra/blob/master/config/jobs/kubernetes-sigs/azuredisk-csi-driver/azuredisk-csi-driver-config.yaml)
  - [x] [azurefile-csi-driver-config.yaml](https://github.com/kubernetes/test-infra/blob/master/config/jobs/kubernetes-sigs/azurefile-csi-driver/azurefile-csi-driver-config.yaml)
  - [x] [blobfuse-csi-driver-config.yaml](https://github.com/kubernetes/test-infra/blob/master/config/jobs/kubernetes-sigs/blobfuse-csi-driver/blobfuse-csi-driver-config.yaml)
  - [x] [dualstack-e2e.yaml](https://github.com/kubernetes/test-infra/blob/master/config/jobs/kubernetes/sig-network/dualstack-e2e.yaml) (cc @aramase)
  - [ ] [sig-windows-config.yaml](https://github.com/kubernetes/test-infra/blob/master/config/jobs/kubernetes-sigs/sig-windows/sig-windows-config.yaml) 

**Provide any links for context**:

/assign
/area provider/azure

### PR DESCRIPTION
**Pull request by chewong**
Fixes #16132

This PR 
- converts all windows job configuration to pod utilities
- moves periodic release branch jobs to separate files
- points --aksengine-download-url of 1.18 jobs to v0.48.0 aks-engine

/assign @marosset @adelina-t

### CODE REVIEW

File: config/jobs/kubernetes-sigs/sig-windows/release-1.14-windows.yaml [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+      - --provider=skeleton
+      - --aksengine-admin-username=azureuser
+      - --aksengine-admin-password=AdminPassw0rd
+      - --aksengine-creds=$(AZURE_CREDENTIALS)
+      # Note: Updating aks-engine version for 1.14 + Windows configs has caused issues in the past - update this config with caution!
+      - --aksengine-download-url=https://github.com/Azure/aks-engine/releases/download/v0.31.2/aks-engine-v0.31.2-linux-amd64.tar.gz
+      - --aksengine-public-key=$(K8S_SSH_PUBLIC_KEY_PATH)
+      - --aksengine-winZipBuildScript=$(WIN_BUILD)
+      - --aksengine-orchestratorRelease=1.14
+      - --aksengine-template-url=https://raw.githubusercontent.com/kubernetes-sigs/windows-testing/master/job-templates/kubernetes_release.json
```
Comment (marosset) [inline config/jobs/kubernetes-sigs/sig-windows/release-1.14-windows.yaml]: Can you update this line to point to the kubernetes_release_1.14.json file we added last week?

**Comment by chewong** [comment]
Rebased the PR. @marosset PTAL. Thanks.

**Pull request merged**